packages/contracts/contracts/Unirep.sol

```diff
@@ -235,6 +241,38 @@ contract Unirep is IUnirep, zkSNARKHelper, VerifySignature {
             revert InvalidSNARKField(AttestationFieldError.GRAFFITI);
     }
 
+    // increment the hashchain and leave the chain unsealed.
+    // Also store a sealed copy of the hashchain in the epoch tree
+    function storeAttestation(
+        Attestation calldata attestation,
+        uint256 epochKey
+    ) public {
```

Comment:
I think it should be private
After an attestation event comes, it update the attestation hash chains.